op-challenger/game/keccak/fetcher/fetcher.go

```diff
@@ -0,0 +1,114 @@
+package fetcher
```

Comment:
The fetcher seems to fit much more clearly alongside the fault contracts, specifically the preimage oracle as opposed to being in the `keccak` package. Especially since we are importing the contracts directly here
(`	"github.com/ethereum-optimism/optimism/op-challenger/game/fault/contracts"`).

Is there a good reason to place it in this package I'm failing to consider?

Reply:
I have been operating off of the, perhaps preconceived, notion that `keccak` package logic should be generic such that it doesn't rely on `fault` package logic and so the `fault` package can freely depend on the `keccak` package.

Reply:
I'm borderline on this but I _think_ this should stay out of the contracts package. It goes beyond just being a mapping layer for the contract to querying more general block data. And it's complex enough that it feels worthy of its own "thing" rather than making the preimage oracle bindings even more complex.

Reply:
That's a fair point. Also a "fetcher" in it of itself should probably live outside the tightly coupled contract binding logic, so I'm ok keeping this in here. I think it's likely that this discussion means we should consider rejigging the package layout at a later date once the logic is worked out/implemented.